orttraining/orttraining/python/training/ortmodule/_options.py

```diff
@@ -0,0 +1,336 @@
+
+    @property
+    def save_onnx_models(self):
+        """Accessor for the ONNX saving configuration."""
+
+        return self._save_onnx_models
+
+    @property
+    def logging(self):
+        """Accessor for the logging configuration."""
+
+        return self._logging
+
+
+class _SkipCheck(IntFlag):
+    """Enumeration to specify which checks should be skipped, allowing faster execution"""
+
+    SKIP_CHECK_DISABLED = 1
+    SKIP_CHECK_DEVICE = 2
+    SKIP_CHECK_BUILD_GRADIENT = 4
+    SKIP_CHECK_EXECUTION_AGENT = 8
+
+    def is_set(self, check):
+        """Check whether `check` is set on the `_SkipCheck instance
+
+        SKIP_CHECK_DISABLED implies the check will return False
+        """
+
+        return not _SkipCheck.is_disabled(self) and check in self
+
+    def is_disabled(self):
+        """Check whether `_SkipCheck.SKIP_CHECK_DISABLED is set on the `_SkipCheck instance"""
+
+        return _SkipCheck.SKIP_CHECK_DISABLED in self
+
+
+class _RuntimeOptions:
+    """Configurable runtime options for ORTModule."""
+
+    def __init__(self, logger: Logger, config=None):
+        """Constructor for _RuntimeOptions.
+
+        Initially set all the options to their default values, then override them with the values
+        from the environment variables.
+        """
+        self._logger = logger
+
+        self.onnx_opset_version = ortmodule.ONNX_OPSET_VERSION
+        self.conv_algo_search = "HEURISTIC"
+
+        # Configuration for cast optimization.
+        # Specify cast propagation strategy. Currently, three strategies are available:
+        #  NONE, INSERT-AND-REDUCE and FLOOD-FILL
+        # The default is FLOOD_FILL, expand FP16 computation regions in the graph using
+        # allowed opcodes for the given level.
+        self._propagate_cast_ops_strategy = C.PropagateCastOpsStrategy.FLOOD_FILL
+        # Optimize by moving Cast operations if propagate_cast_ops_level is non-negative.
+        # - If the _propagate_cast_ops_level is set to zero, then the transformation considers only the opcodes
+        #   specified by _propagate_cast_ops_allow as "FP16 safe", to insert/(re)move cast operations before/after
+        #   to perform such operations in reduced (16-bit) precision.
+        # - If propagate_cast_ops_level is positive, 1 or 2, then in addition to opcode codes specified by
+        #   propagate_cast_ops_allow, use onnxruntime predetermined list of opcodes considered safe to move
+        #   before/after the cast operation.
+        # - Onnxruntime Level 1 predetermined "FP16 safe" opcodes include only opcodes that do not perform
+        #   any computation such as Transpose, Split, Reshape, etc., or the computation is actually in Float
+        #   such as GeLU, etc.
+        # - Whereas Level 2 predetermined "FP16 safe" opcodes include opcodes that perform computation using
+        #   contrib ops, Dropout, LayerNormalization, etc.
+        self._propagate_cast_ops_level = 1
```

Comment:
Considering the class is already internal (`_RuntimeOptions`), shall we name the members without the `_` prefixed?